ui-core/src/components/inputs/datePicker/useDatePicker.tsx

```diff
@@ -0,0 +1,100 @@
+import useClickOutside from '../../hooks/useOutsideClick';
+import { useModalPosition } from '../../hooks/useModalPosition';
+import { CalendarSlot } from './type';
+import { formatDateString } from './utils';
+import { DatePickerProps } from './DatePicker';
+
+const MODAL_HORIZONTAL_OFFSET = -24;
+const MODAL_VERTICAL_OFFSET = 3;
+const INPUT_FORMAT = 'dd/mm/yy';
+
+export default function useDatePicker({ calendarPickerProps, inputProps }: DatePickerProps) {
+  const [showPicker, setShowPicker] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+  const [selectedSlot, setSelectedSlot] = useState<CalendarSlot | undefined>(
+    calendarPickerProps.selectedSlot
+  );
+
+  const calendarPickerRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  useClickOutside(calendarPickerRef, (e) => {
+    // Do not close the picker if any children in input wrapper is clicked.
+    // This wrapper include, the input itself, the trailing content (which contains the calendar icon) and the leading content
+    if (inputRef.current && inputRef.current.parentElement?.contains(e.target as Node)) return;
+    setShowPicker(false);
+  });
+  const { calculatePosition, modalPosition } = useModalPosition(
+    inputRef,
+    calendarPickerRef,
+    MODAL_VERTICAL_OFFSET,
+    MODAL_HORIZONTAL_OFFSET
+  );
+
+  const isSingleMode = calendarPickerProps.mode === 'single';
+  const isRangeMode = calendarPickerProps.mode === 'range';
+
+  const formatSlotToInputValue = (newSelectedSlot: CalendarSlot | undefined) => {
+    let formattedDate = '';
+
+    if (newSelectedSlot) {
+      if (isSingleMode && newSelectedSlot.start) {
+        formattedDate = formatDateString(newSelectedSlot.start);
+      }
+
+      if (isRangeMode) {
+        const { start, end } = newSelectedSlot;
+        if (start && end) {
+          formattedDate =
+            start.getFullYear() === end.getFullYear()
+              ? `${formatDateString(start).slice(0, 5)} - ${formatDateString(end)}`
+              : `${formatDateString(start)} - ${formatDateString(end)}`;
+        }
+
+        if (start && !end) {
+          formattedDate = `${formatDateString(start)} -`;
+        }
+      }
+    }
+    return formattedDate;
+  };
+
+  const handleCalendarPickerChange = (newSelectedSlot: CalendarSlot | undefined) => {
+    setSelectedSlot(newSelectedSlot);
+    const formattedDate = formatSlotToInputValue(newSelectedSlot);
+    setInputValue(formattedDate);
+  };
+
+  const handleInputClick = (e: React.MouseEvent<HTMLInputElement, MouseEvent>) => {
+    setShowPicker(true);
+    inputProps.onClick?.(e);
+  };
+
+  useEffect(() => {
+    if (showPicker) {
+      calculatePosition();
+    }
+  }, [showPicker, calculatePosition]);
+
+  const setRefs = (el: HTMLInputElement | null) => {
+    if (el) {
+      inputRef.current = el;
+      withMask('datetime', {
+        inputFormat: INPUT_FORMAT,
+      });
```

Comment:
We wanted to use this method to handle characters entered by the user. As mentionned in the [docs](https://github.com/eduardoborges/use-mask-input?tab=readme-ov-file#usage-with-react-final-form) It was supposed to apply the mask defined in `INPUT_FORMAT` to format the value entered. But we don't need this anymore because for this first version, user can set the input value only using the calendar picker.